cumulusci/tasks/bulkdata/generate_and_load_data.py

```diff
@@ -0,0 +1,110 @@
+
+    task_docs = """
+    Orchestrate creating tempfiles, generating data, loading data, cleaning up tempfiles and batching.
+
+    CCI has features for generating data and for loading them into orgs. This task pulls them
+    together to give some useful additional features, such as storing the intermediate data in
+    a tempfile (the default behavior) and generating the data in batches instead of all at
+    once (controlled by the `batch_size` option).
+
+    The simplest possible usage is to specify the number of records you'd like generated, a
+    mapping file that defines the schema and a data generation task written in Python to actually
+    generate the data.
+
+    Use the `num_records` option to specify how many records to generate.
+    Use the `mapping` option to specify a mapping file.
+    Use 'data_generation_task' to specify what Python class to use to generate the data.'
+    Use 'batch_size' to specify how many records to generate and upload in every batch.
+
+    By default it creates the data in a temporary file and then cleans it up later. Specify database_url if you
+    need more control than that. The use of both database_url and batch_size together is not currently supported.
+    """
+
+    task_options = {
+        "num_records": {
+            "description": "How many records to generate. Precise calcuation depends on the generator.",
+            "required": True,
+        },
+        "batch_size": {
+            "description": "How many records to create and load at a time.",
+            "required": False,
+        },
+        "mapping": {"description": "A mapping YAML file to use", "required": True},
+        "data_generation_task": {
+            "description": "Fully qualified class path of a task to generate the data. Look at cumulusci.tasks.bulkdata.tests.dummy_data_factory to learn how to write them.",
+            "required": True,
+        },
+        "data_generation_options": {
+            "description": "Options to pass to the data generator.",
+            "required": False,
+        },
+        "database_url": {
+            "description": "A URL to store the database (defaults to a transient SQLite file)",
+            "required": "",
+        },
+    }
+
+    def _run_task(self):
+        mapping_file = os.path.abspath(self.options["mapping"])
+        assert os.path.exists(mapping_file), f"{mapping_file} cannot be found."
+        database_url = self.options.get("database_url")
+        num_records = int(self.options["num_records"])
+        batch_size = int(self.options.get("batch_size", num_records))
+        class_path = self.options.get("data_generation_task")
+        self.data_generation_task = import_global(class_path)
+
+        if database_url and batch_size != num_records:
+            raise TaskOptionsError(
+                "You may not specify both `database_url` and `batch_size` options."
+            )
```

Comment:
It would be best to do all the initialization of options in the _init_options method. That way it'll fail early while initializing a flow that uses the task instead of waiting until the task is running. (I know you're not using this as part of a flow in this case, but that's the general motivation for separating the initialization of options.)

Reply:
https://github.com/SFDO-Tooling/CumulusCI/blob/feature/generate-and-load-task/cumulusci/tasks/bulkdata/generate_and_load_data.py#L59-L74